spec/index.html

```diff
@@ -10424,7 +10424,205 @@ <h3>Evaluation Semantics</h3>
+              </p>
+              <table style="border-collapse: collapse; border-color: #000000; border-spacing:5px; border-width: 1px">
+                <tbody>
+                  <tr>
+                    <th>Syntax Form</th>
+                    <th>In-scope variables</th>
+                  </tr>
+                  <tr>
+                    <td>`EXISTS` and `NOT EXISTS` filters </td>
+                    <td><code>v</code> is in-scope if it is in-scope for the pattern to which the `FILTER` is applied.
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+              <div class="note">
+                <p>
+                  This restriction means that <a href="#defn_valuesinsertion">values inserted</a>
+                  do not conflict with values assigned to variables within the pattern.
+                </p>
+                <p>
+                  This operation is performed as part of query parsing.
+                </p>
+              </div>
+            </section>
+            <section>
+              <h4>Remapping</h4>
+              <p>
+                Remapping ensures that a variable name used inside a project expression,
+                but which is not part of the results of the evaluation of the project expression,
+                does not coincide with a variable mentioned anywhere else in the
+                algebra expression of a query.
+              </p>
+              <p>
+                Renaming these variables does not change the results of evaluating
+                the project expresssion.
+              </p>
+              <div class="defn">
+                <b>Definition: <span id="defn_projmap" name="defn_projmap">Projection Expression Variable Remapping</span></b>
+                <p>
+                  For a projection algebra operation #sparqlProjection `Project(A, PV)` acting on algreg express `A` and with set of variables `PV`, define
+                  a partial mapping `F` from
+                  `<a href="#sparqlQueryVariables">V</a>`,
+                  the set of all variables, to `V` where:
+                </p>
+                <pre>F(v) = v1 if v is in PV, where v1 is a fresh variable
+F(v) = v if v is not in PV</pre>
+                <p>
+                Define the Projection Expression Variable Remapping `ProjectMap(P, PV)`
+                </p>
+                <pre>ProjectMap(Project(A, PV)) = Project(A1, PV) 
+                                where A1 is the result of applying F
+                                to every variable mentioned in A.
+                </pre>
+                <p>
+                  The Projection Expression Variable Remapping yields an algrebra expression that
+                  evaluates to the same results as the Project argument. No variable of `ProjectMap(Project(A, PV))`
+                  that is not in `PV` is mentioned anywhere else in the algebra expression for the query.
```

Comment:
I don't think that's true based on the current version of this definition. By the current definition, the projection variables are renamed from `A` to `A1`, but not in `PV`. As a consequence, the evaluation of `A1` will not even produce bindings for the projection variables.

Is it possible that you have a mistake in the earlier part of this definition? In particular, I have the feeling that the first line of the definition of `F` should have been as follows.
```
F(v) = v1 if v is mentioned in A but is not in PV , where v1 is a fresh variable
```

Reply:
Indeed - the definition of F(v) was inverted.